packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js

```diff
@@ -63,35 +115,87 @@ const sumDataValues = (analytics, dataValues, filter = {}) => {
   return sum;
 };
 
-const countDataValues = (analytics, dataValues, filter, config) => {
+const countDataValues = async (analytics, dataValues, filter, config, models) => {
   return sumDataValues(
     analytics.map(a => ({
       ...a,
       value: checkValueSatisfiesCondition(a.value, config?.countCondition || '*') ? 1 : 0,
     })),
     dataValues,
     filter,
+    models,
   );
 };
 
+const buildOrganisationUnitParentValue = async (analytics, models) => {
+  const orgUnitCodes = [...new Set(analytics.map(a => a.organisationUnit))];
+  const parentIdMap = {};
+  const orgUnitParentMap = {};
+
+  const orgUnits = await models.entity.find({ code: orgUnitCodes });
+  const orgUnitParentIds = [...new Set(orgUnits.map(o => o.parent_id))];
+  const orgUnitParents = await models.entity.find({ id: orgUnitParentIds });
+  orgUnitParents.forEach(p => {
+    parentIdMap[p.id] = { code: p.code, type: p.type };
+  });
```

Comment:
Better to keep all parent fields - allows for more filtering options in the future:

```suggestion
  const parentIdMap = keyBy(orgUnitParents, 'id');
```

Reply:
I hear you. The other consideration using aggregation on ancestors was that we want only immediate ancestors (parents) for aggregation. Which could be achieved by adding a parameter or creating a new aggregation type? Coupled with the aggregating on data values as well, climbing through the broken window seemed path of least pain.